pkg/adaptor/mongodb/reader.go

```diff
@@ -140,7 +140,7 @@ func (r *Reader) iterateCollection(mgoSession *mgo.Session, in <-chan string, do
 func (r *Reader) catQuery(c string, lastID interface{}, mgoSession *mgo.Session) *mgo.Query {
 	query := bson.M{}
 	if lastID != nil {
-		query = bson.M{"_id": bson.M{"$gte": lastID}}
+		query = bson.M{"_id": bson.M{"$gt": lastID}}
```

Comment:
I'm not sure about the caching, but I can see that since `lastID` is updated only if there is no iterator error when reading, `$gt` is correct. Nice catch.